src/org/labkey/test/tests/FileAttachmentColumnTest.java

```diff
@@ -15,153 +15,609 @@
+            doAndWaitForDownload(()-> downloadLink.click());
+        }
 
         // verify popup/sprite for jpeg
         mouseOver(Locator.tagWithAttribute("img", "title", SAMPLE_JPG.getName()));
         shortWait().until(ExpectedConditions.presenceOfElementLocated(By.xpath("//div/span[contains(text(),'" + SAMPLE_JPG.getName() + "')]")));
         mouseOut();
     }
 
-    @Override
-    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    // sampleType
+    @Test
+    public void testSampleFileFields()
     {
-        _containerHelper.deleteProject(getProjectName(), afterTest);
+        SampleTypeDefinition subASampleType = new SampleTypeDefinition(SUBA_SAMPLETYPE_NAME)
+                .setFields(List.of(new FieldDefinition("color", ColumnType.String),
+                        new FieldDefinition("file", ColumnType.File)));
+        SampleTypeAPIHelper.createEmptySampleType(SUBFOLDER_A_PATH, subASampleType);
+
+        // give the sampleType actual files by adding individual rows; for the nonce not all files are resolving on import
+        SampleTypeHelper.beginAtSampleTypesList(this, SUBFOLDER_A_PATH);
+        clickAndWait(Locator.linkWithText(SUBA_SAMPLETYPE_NAME));
+        DataRegionTable samplesRegion = new DataRegionTable("Material", getDriver());
+        List<String> fileNames = SAMPLE_FILES.stream().map(File::getName).toList();
+
+        for (File file : SAMPLE_FILES)
+        {
+            var queryUpdatePage = samplesRegion.clickInsertNewRow();
+            queryUpdatePage.setField("Name", file.getName());
+            queryUpdatePage.setField("file", file)
+                    .submit();
+        }
+        validateSampleData(SUBA_SAMPLETYPE_NAME, SUBFOLDER_A_PATH, SAMPLE_FILES);
+
+        var fileContentPage = FileContentPage.beginAt(this, SUBFOLDER_A_PATH);
+        fileContentPage.fileBrowserHelper().selectFileBrowserItem("/sampletype/csv_sample.csv");
+        var files = fileContentPage.fileBrowserHelper().getFileList(true);
+        checker().withScreenshot("unexpected_davfile_state")
+                .wrapAssertion(()-> Assertions.assertThat(files)
+                        .as("expect edited files to be present in dav locaiton")
+                        .containsExactlyInAnyOrderElementsOf(fileNames));
     }
 
-    @BeforeClass
-    public static void setupProject()
+    // assay
+    @Test
+    public void testAssayFileFieldsUI() throws Exception
     {
-        FileAttachmentColumnTest init = (FileAttachmentColumnTest)getCurrentTest();
-        init.doSetup();
+        String runName = "assay_run.tsv";
+
+        // generate some resultFiles
+        var resultFiles = new ArrayList<File>();
+        WebDavUploadHelper uploadHelper = new WebDavUploadHelper(SUBFOLDER_A_PATH);
+        List<Map<String, Object>> importData = new ArrayList<>();
+        for (int i = 0; i < 5; i++)
+        {
+            String fileName = String.format("field_file_for_results_domain-%d.tsv", i);
```

Comment:
Why using 5 different result files? This seems they could be 5 data rows in a single result file. Or, have 1 result file that has N rows, and another result file with 1 row.